test/test-parse-warning.cpp

```diff
@@ -390,6 +390,68 @@ TEST_F(test_parse_warning,
   }
 }
 
+TEST_F(test_parse_warning, warn_on_comma_operator_in_conditional_statement) {
+  {
+    test_parser p(u8"if(false, true){}"_sv, capture_diags);
+    p.parse_and_visit_statement();
+    EXPECT_THAT(
+        p.errors,
+        ElementsAreArray({
+            DIAG_TYPE_OFFSETS(
+                p.code, diag_misleading_comma_operator_in_conditional_statement,
+                comma, strlen(u8"if(false"), u8","_sv),
+        }));
+  }
+
+  {
+    test_parser p(u8"do{i++}while((i < 0), true)"_sv, capture_diags);
```

Comment:
Are the parentheses necessary to trigger the warning? Could we write the test like this instead:
```suggestion
    test_parser p(u8"do{i++}while(i < 0, true)"_sv, capture_diags);
```